crates/schema-inference/src/inference.rs

```diff
@@ -0,0 +1,178 @@
+use doc::inference::{ArrayShape, ObjProperty, ObjShape, Shape};
+use json::schema::types;
+use serde_json::Value as JsonValue;
+
+use crate::shape;
+
+pub fn infer_shape(value: &JsonValue) -> Shape {
+    match value {
+        JsonValue::Bool(value) => infer_bool(value),
+        JsonValue::Number(value) => infer_number(value),
+        JsonValue::String(value) => infer_string(value.as_ref()),
+        JsonValue::Null => infer_null(),
+        JsonValue::Array(inner) => infer_array(inner.as_slice()),
+        JsonValue::Object(values) => infer_object(values),
+    }
+}
+
+fn infer_bool(_value: &bool) -> Shape {
+    Shape {
+        type_: types::BOOLEAN,
+        ..Default::default()
+    }
+}
+
+fn infer_number(value: &serde_json::Number) -> Shape {
+    let type_ = if value.is_f64() {
+        types::FRACTIONAL
+    } else {
+        types::INTEGER
+    };
+
+    Shape {
+        type_,
+        ..Default::default()
+    }
+}
+
+fn infer_string(_value: &str) -> Shape {
+    Shape {
+        type_: types::STRING,
```

Comment:
No action now, but I just wanted to note that I suspect that in the future we may want to try to infer some string formats (i.e. `date-time`) so that those values can have proper column types when we materialize them. I agree with your decision _not_ to worry about that right now, though.